File: SWEA/D2/1974. 스도쿠 검증/스도쿠 검증.py
t = int(input())

for test_case in range(1, t+1):
    board = []
    for _ in range(9):
        temp = list(map(int,input().split()))
        board.append(temp)
    
    flag = True
    # check row
    for i in range(9):
        temp = set()
        for number in board[i]:
            temp.add(number)
        if len(temp) != 9:
            flag = False
    # check column
    if flag:
        for i in range(9):
            temp = set()
            for j in range(9):
                temp.add(board[j][i])
            if len(temp) != 9:
                flag = False
    # check square
    if flag:
        for i in range(0,9,3):
            for j in range(0,9,3):
                temp = set()
                for k in range(i, i+3):
                    for l in range(j, j+3):
                        temp.add(board[l][k])
                if len(temp) != 9:
                    flag = False
    
    if flag:
        print("#{} {}".format(test_case, 1))
    else:
        print("#{} {}".format(test_case, 0))

File: SWEA/D2/1974. 스도쿠 검증/스도쿠 검증.java
import java.util.*;
import java.io.*;

class Solution
{
    static int[] dx = {1, 0, -1, 0};
    static int[] dy = {0, 1, 0, -1};
    public static void main(String args[]) throws Exception
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int T = Integer.parseInt(br.readLine());

        for(int testCase = 1; testCase <= T; testCase++) {

            int[][] board = new int[9][9];
            for(int i=0; i<9; i++) {
                StringTokenizer st = new StringTokenizer(br.readLine());
                for(int j=0; j<9; j++) {
                    board[i][j] = Integer.parseInt(st.nextToken());
                }
            }

            boolean flag = true;

            // check row & col
            for (int i = 0; i < 9; i++) {
                int sum1 = 0;
                int sum2 = 0;
                for (int j = 0; j < 9; j++) {
                    sum1 += board[i][j];
                    sum2 += board[j][i];
                }
                if (sum1 != 45 || sum2 != 45) {
                    flag = false;
                    break;
                }
            }

            // check square
            if (flag) {
                for(int sr = 0; sr < 9; sr+=3) {
                    for(int sc = 0; sc < 9; sc+=3) {
                        int sum = 0;
                        for(int i=sr; i<sr + 3; i++) {
                            for(int j=sc; j<sc + 3; j++) {
                                sum += board[i][j];
                            }
                        }
                        if (sum != 45) {
                            flag = false;
                            break;
                        }
                    }
                    if (!flag) break;
                }
            }

            sb.append("#").append(testCase).append(" ")
                    .append(flag ? 1 : 0).append("\n");
        }

        System.out.println(sb);
    }
}